src/jaxsim/rbda/contacts/relaxed_rigid.py

```diff
@@ -0,0 +1,356 @@
+            and jnp.all(self.max_iterations > 0)
+            and jnp.all(self.tolerance > 0.0)
+        )
+
+
+@jax_dataclasses.pytree_dataclass
+class RelaxedRigidContactsState(ContactsState):
+    """Class storing the state of the relaxed rigid contacts model."""
+
+    def __eq__(self, other: RelaxedRigidContactsState) -> bool:
+        return hash(self) == hash(other)
+
+    @staticmethod
+    def build() -> RelaxedRigidContactsState:
+        """Create a `RelaxedRigidContactsState` instance"""
+
+        return RelaxedRigidContactsState()
+
+    @staticmethod
+    def zero(model: js.model.JaxSimModel) -> RelaxedRigidContactsState:
+        """Build a zero `RelaxedRigidContactsState` instance from a `JaxSimModel`."""
+        return RelaxedRigidContactsState.build()
+
+    def valid(self, model: js.model.JaxSimModel) -> bool:
+        return True
+
+
+@jax_dataclasses.pytree_dataclass
+class RelaxedRigidContacts(ContactModel):
+    """Rigid contacts model."""
+
+    parameters: RelaxedRigidContactsParams = dataclasses.field(
+        default_factory=RelaxedRigidContactsParams
+    )
+
+    terrain: jax_dataclasses.Static[Terrain] = dataclasses.field(
+        default_factory=FlatTerrain
+    )
+
+    def compute_contact_forces(
+        self,
+        position: jtp.Vector,
+        velocity: jtp.Vector,
+        model: js.model.JaxSimModel,
+        data: js.data.JaxSimModelData,
+    ) -> tuple[jtp.Vector, tuple[Any, ...]]:
+
+        def _detect_contact(x: jtp.Array, y: jtp.Array, z: jtp.Array) -> jtp.Array:
+            x, y, z = jax.tree_map(jnp.squeeze, (x, y, z))
+
+            n̂ = self.terrain.normal(x=x, y=y).squeeze()
+            h = jnp.array([0, 0, z - model.terrain.height(x=x, y=y)])
+
+            return jnp.dot(h, n̂)
+
+        # Compute the activation state of the collidable points
+        δ = jax.vmap(_detect_contact)(*position.T)
+
+        with data.switch_velocity_representation(VelRepr.Mixed):
+            M = js.model.free_floating_mass_matrix(model=model, data=data)
+            J_WC = jnp.vstack(
+                jax.vmap(lambda J, height: J * (height < 0))(
+                    js.contact.jacobian(model=model, data=data)[:, :3], δ
+                )
+            )
+            W_H_C = js.contact.transforms(model=model, data=data)
+            h = js.model.free_floating_bias_forces(model=model, data=data)
+            W_ν = data.generalized_velocity()
+            J̇_WC = jnp.vstack(
+                jax.vmap(lambda J̇, height: J̇ * (height < 0))(
+                    js.contact.jacobian_derivative(model=model, data=data)[:, :3], δ
+                ),
+            )
+
+            a_ref, R, K, D = self._regularizers(
+                model=model,
+                penetration=δ,
+                velocity=velocity,
+                parameters=self.parameters,
+            )
+
+        M_inv = jnp.linalg.inv(M)
+        G = J_WC @ M_inv @ J_WC.T
+        CW_al_free_WC = J_WC @ M_inv @ (-h) + J̇_WC @ W_ν
```

Comment:
Shouldn't this be computed with $\dot{\boldsymbol{\nu}}_{\text{free}}$ (computed e.g. with [`js.ode.system_acceleration`](https://github.com/ami-iit/jaxsim/blob/cd6cb10548b2c87f49fba7d49741f2b46e1296d2/src/jaxsim/api/ode.py#L191)) like it was done [for the rigid contacts](https://github.com/ami-iit/jaxsim/blob/cd6cb10548b2c87f49fba7d49741f2b46e1296d2/src/jaxsim/rbda/contacts/rigid.py#L441)?

Reply:
Yes, thanks! Done in https://github.com/ami-iit/jaxsim/pull/223/commits/89843345142c7d73b88be31940d376cf947bcf38